### PR TITLE
SRM - Grafana Loki health source

### PR DESCRIPTION
**Pull request by SudheendraKatte**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/service-reliability-management/monitored-service/health-source/loki.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+Currently, this feature is behind the feature flag `SRM_ENABLE_GRAFANA_LOKI_LOGS`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
+:::
+
+
+In Harness, a health source refers to a mapping that establishes a connection between a service within Harness and a service operating in a deployment environment. This deployment environment is under the monitoring of an Application Performance Monitoring (APM) or logging tool. Through this mapping, Harness is able to gather metrics and data from the APM or logging tool, utilizing it to assess the health and status of the respective service within Harness.
+
+
+## Prerequisites
+
+- Grafana Loki connector has been added to the Harness platform. To know how to add the Grafana Loki connector, go to [Add Grafana Loki](/docs/platform/Connectors/Monitoring-and-Logging-Systems/connect-to-monitoring-and-logging-systems#add-grafana-loki)
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/health-source/loki.md:23]:
```suggestion
- A Grafana Loki connector in the Harness platform. To learn how to add a Grafana Loki connector, go to [Add Grafana Loki](/docs/platform/Connectors/Monitoring-and-Logging-Systems/connect-to-monitoring-and-logging-systems#add-grafana-loki).
```

File: docs/service-reliability-management/monitored-service/health-source/loki.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+- A monitored service has already been created in the Harness SRM. If you haven't yet created one, go to [Create a monitored service](/docs/service-reliability-management/monitored-service/create-monitored-service)
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/health-source/loki.md:25]:
```suggestion
- A monitored service created in the Harness SRM. If you haven't yet created one, go to [Create a monitored service](/docs/service-reliability-management/monitored-service/create-monitored-service).
```

File: docs/service-reliability-management/monitored-service/health-source/loki.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+
+## Add Grafana Loki health source
+
+To add Grafana Loki as a health source:
+
+1. In your Harness project, go to **Service Reliability** > **Monitored Services**.  
+   A list of monitored services is displayed.
+
+2. Locate the monitored service for which you want to add Grafana Loki health source, select the three vertical dots next to it, and then select **Edit service**.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/health-source/loki.md:35]:
```suggestion
2. Locate the monitored service for which you want to add a Grafana Loki health source, select the three vertical dots next to it, and then select **Edit service**.  
```

File: docs/service-reliability-management/monitored-service/health-source/loki.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+   The Configurations page appears.
+
+3. Go to the **Service** tab, and under **Define Your Sources**, select **+ Add New Health Source**.
+   
+   The Add New Health Source page appears.
+
+
+### Define a health source
+
+On the Define Health Source page of the Add New Health Source page, do the following:
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/health-source/loki.md:45]:
```suggestion
On the Define Health Source section of the Add New Health Source page, do the following: 
```

File: docs/service-reliability-management/monitored-service/health-source/loki.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+2. Enter a name for the health source.
+   
+3. Under **Connect Health Source**, select **Select Connector**.
+   
+4. In the Create or Select an Existing Connector dialog, select a Grafana Loki connector, and then select **Apply Selected**.
+
+   <details>
+   <summary><b>Follow these steps to create a new Grafana Loki connector.</b></summary>
+
+      Follow these steps to create a new Grafana Loki connector.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/health-source/loki.md:58]:
```suggestion
```

File: docs/service-reliability-management/monitored-service/health-source/loki.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+      1.	In the Create or Select an Existing Connector dialog, select **+ New Connector**.
+
+      1. In the **Overview** tab, enter a name for the connector, an optional description, and a tag, and then select **Continue**. If you are going to use multiple providers of the same type, ensure you give each provider a different name.
+ 
+      2. In the **Headers** tab, enter the following and select **Next**:
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/health-source/loki.md:64]:
```suggestion
      2. In the **Headers** tab, enter the following, and select **Next**:
```

File: docs/service-reliability-management/monitored-service/health-source/loki.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+      3. **Base URL** of your Grafana Loki account. By default, Grafana Loki exposes its API on the 3100 port without any authentication.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/health-source/loki.md:65]:
```suggestion
         - **Base URL** of your Grafana Loki account. By default, Grafana Loki exposes its API on the 3100 port without any authentication. 
```

File: docs/service-reliability-management/monitored-service/health-source/loki.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+
+      Follow these steps to create a new Grafana Loki connector.
+
+      1.	In the Create or Select an Existing Connector dialog, select **+ New Connector**.
+
+      1. In the **Overview** tab, enter a name for the connector, an optional description, and a tag, and then select **Continue**. If you are going to use multiple providers of the same type, ensure you give each provider a different name.
+ 
+      2. In the **Headers** tab, enter the following and select **Next**:
+      3. **Base URL** of your Grafana Loki account. By default, Grafana Loki exposes its API on the 3100 port without any authentication. 
+      4. Optionally, the **Key** and **Value** pair for the Grafana Loki log stream that you want to select in the query. For more information on the log stream selector and key-value pairs, go to [Log stream selector](https://grafana.com/docs/loki/latest/logql/log_queries/#log-stream-selector).
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/health-source/loki.md:66]:
```suggestion
         - Optionally, the **Key** and **Value** pair for the Grafana Loki log stream that you want to select in the query. For more information on the log stream selector and key-value pairs, go to [Log stream selector](https://grafana.com/docs/loki/latest/logql/log_queries/#log-stream-selector).
```

File: docs/service-reliability-management/monitored-service/health-source/loki.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+   
+      5.  Optionally, in the **Parameters** tab, enter the **Key** and **Value** pair.
+   
+      6.  In the **Validation Path** tab, select either the **GET** or **POST** request method and enter the **Validation Path**.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/health-source/loki.md:70]:
```suggestion
      6.  In the **Validation Path** tab, select either the **GET** or **POST** request method, and enter the **Validation Path**.  
```

File: docs/service-reliability-management/monitored-service/health-source/loki.md [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+    
+### Define log configuration settings
+
+1. In the **Configuration** tab of the Add New Health Source page, select **+ Add Query**.  
+   
+   The Add Query dialog appears.
+
+2. Enter a name for the query and then select **Submit**.  
+   
+   The query that you added gets listed under **Logs Group**. The query specification and mapping settings are displayed.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/health-source/loki.md:102]:
```suggestion
   The query that you added is listed under **Logs Group**. The query specification and mapping settings are displayed.
```

File: docs/service-reliability-management/monitored-service/health-source/loki.md [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+   These settings help you retrieve the desired logs from the Grafana Loki platform and map them to the Harness service. 
+
+
+#### Define a query
+   
+1. In the **Query** field, enter a log query and select **Run Query** to execute it.
```
Comment (brian-f-harness) [inline docs/service-reliability-management/monitored-service/health-source/loki.md:108]:
```suggestion
1. In the **Query** field, enter a log query, and select **Run Query** to execute it.
```

**Review by brian-f-harness — APPROVED** [review]
lgtm. Please check the numbering under Follow these steps to create a new Grafana Loki connector.

File: docs/service-reliability-management/monitored-service/health-source/loki.md (hunk 1 above)
Comment (SudheendraKatte) [reply docs/service-reliability-management/monitored-service/health-source/loki.md:23]: Updated

File: docs/service-reliability-management/monitored-service/health-source/loki.md (hunk 2 above)
Comment (SudheendraKatte) [reply docs/service-reliability-management/monitored-service/health-source/loki.md:25]: Updated

File: docs/service-reliability-management/monitored-service/health-source/loki.md (hunk 3 above)
Comment (SudheendraKatte) [reply docs/service-reliability-management/monitored-service/health-source/loki.md:35]: Updated

File: docs/service-reliability-management/monitored-service/health-source/loki.md (hunk 4 above)
Comment (SudheendraKatte) [reply docs/service-reliability-management/monitored-service/health-source/loki.md:45]: Upadated

File: docs/service-reliability-management/monitored-service/health-source/loki.md (hunk 5 above)
Comment (SudheendraKatte) [reply docs/service-reliability-management/monitored-service/health-source/loki.md:58]: Good catch!

File: docs/service-reliability-management/monitored-service/health-source/loki.md (hunk 7 above)
Comment (SudheendraKatte) [reply docs/service-reliability-management/monitored-service/health-source/loki.md:65]: Corrected

File: docs/service-reliability-management/monitored-service/health-source/loki.md (hunk 8 above)
Comment (SudheendraKatte) [reply docs/service-reliability-management/monitored-service/health-source/loki.md:66]: Corrected

File: docs/service-reliability-management/monitored-service/health-source/loki.md (hunk 9 above)
Comment (SudheendraKatte) [reply docs/service-reliability-management/monitored-service/health-source/loki.md:70]: Updated

File: docs/service-reliability-management/monitored-service/health-source/loki.md (hunk 10 above)
Comment (SudheendraKatte) [reply docs/service-reliability-management/monitored-service/health-source/loki.md:102]: Updated

File: docs/service-reliability-management/monitored-service/health-source/loki.md (hunk 11 above)
Comment (SudheendraKatte) [reply docs/service-reliability-management/monitored-service/health-source/loki.md:108]: Updated

**Comment by SudheendraKatte** [comment]
Incorporated review comments.

**Pull request merged**